### ISSUE TITLE
feat: Improve console UI for apps with flavors

### ISSUE THREAD

**Issue by erickzanardo** [shorebirdtech/shorebird#1991]
**Description**

When using flavor shorebird creates one app per flavor. which is fine. But when an app have too many, the app list on the web console gets polluted.

We could make some UI improvements and group them visually, making the listing more organized.

**Comment by bryanoltman** [comment]
Bumping this: https://discord.com/channels/1030243211995791380/1296141504288395345

**Comment by bryanoltman** [comment]
One related issue: there's no way to get a shorebird.yaml for a flavored app from the console, and it's tricky to regenerate one for a flavored app if some of the flavors (but not all) have been deleted.

**Comment by eseidel** [comment]
Yeah, we need some sort of grouping for apps.  Whatever we call it internally is immaterial, but apps with flavors should show up as a single entity inside our console when possible.
